main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala

```diff
@@ -75,37 +76,69 @@ object BoolUnification {
       case _ => // nop
     }
 
-    // translate the types into formulas
-    implicit val alg: BoolAlg[BoolFormula] = BoolFormula.AsBoolAlg
-
-    val env = alg.getEnv(List(tpe1, tpe2))
-    val f1 = alg.fromType(tpe1, env)
-    val f2 = alg.fromType(tpe2, env)
-
-    val renv = alg.liftRigidityEnv(renv0, env)
+    val typeVars = tpe1.typeVars ++ tpe2.typeVars
+    val varThreshold = 4
```

Comment:
In #5029 I add a new flag `--Xbdd-threshold` which can be accessed as `flix.options.xbddthreshold`.